SQL_Assignments/sql_2.sql

```diff
@@ -0,0 +1,137 @@
+
+    select firstname,lastname from emp where FirstName not like 'john'
+Identify Marketing department employees earning $60,000 or less who were hired after June 30, 2019.
+
+    select firstname,lastname from emp  where Department ='marketing' and Salary <70000 and HireDate> '06-30-2019'
+
+Find employees whose first name contains the letters 'an' anywhere and ends with 'e'.
+
+    select firstname,lastname from emp  where FirstName like '%an%e'
+
+
+
+
+
+
+
+
+CREATE TABLE Sales (
+    SaleID INT PRIMARY KEY,
+    ProductID INT,
+    CustomerID INT,
+    SaleDate DATE,
+    Quantity INT,
+    UnitPrice DECIMAL(10, 2)
+);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY,
+    ProductName VARCHAR(100),
+    Category VARCHAR(50)
+);
+
+CREATE TABLE Customers (
+    CustomerID INT PRIMARY KEY,
+    CustomerName VARCHAR(100),
+    City VARCHAR(50),
+    State VARCHAR(2)
+);
+
+
+
+
+
+
+    insert into customers values(1000,'Abhishek','Trivandrum','kerala')
+
+
+Calculate the total sales amount for each product. Display the ProductID and total sales amount.
+
+    select productid,sum(quantity*unitprice) as total_sales from totalsales group by productid
+
+
+Find the average quantity sold per sale for each product category.
+
+
+    select p.category,avg(quantity) as average from products p,totalsales s where p.productid =s.productid group by p.category
+
+List the top 5 customers by their total purchase amount. Include the CustomerID and total purchase amount.
+
+    select customerid,(quantity*unitprice) as total_purchase from sales order by (quantity*unitprice) desc offset 0  rows fetch next 5 rows only
+
+Determine the number of sales made each month in the year 2023. Display the month and the count of sales.
+    
+    select month(saledate) as monthofsale,count(quantity) as total_sales from sales where year(saledate)=2024 group by month(saledate)
+
+
+Calculate the total revenue for each state, but only for states with more than $10,000 in total sales.
+
+    select sum(quantity*unitprice)as sales,state from totalsales s,customers c where s.customerid=c.customerid group by state having sum(quantity*unitprice)>10000
+
+
+
+Find products that have been sold more than 100 times in total. Display the ProductID, ProductName, and the total quantity sold.
+
+    select p.productname,p.productid,sum(quantity) as total_quantity from products p,totalsales s where s.productid=p.productid group by p.productid,p.productname having sum(quantity)>100
+
+
+
+For each customer, find the date of their first purchase and the date of their most recent purchase.
+
+    select c.customerid,c.customername ,min(saledate) as firstpurchase,max(saledate) as lastpurchase from customers c,totalsales s where s.customerid=c.customerid group by c.customerid,c.customername
+
+
+
+Find the customers who have made purchases on at least 5 different dates. Display the CustomerID and the count of distinct purchase dates.
```

Comment:
How did you verified the output of this assignment ?

Reply:
my table has a customer who has purchased in 3 different dates so i just updated the last part of the query to>2 to check if the query works or not and this works fine so this query will work for 5 as well